components/xip-patch/xip-patch.sh

```diff
@@ -2,77 +2,112 @@
 
 set -o errexit
 
-if [ -z "${INGRESSGATEWAY_SERVICE_NAME}" ]; then
-  INGRESSGATEWAY_SERVICE_NAME=istio-ingressgateway
-fi
-
-if [ -z "${EXTERNAL_PUBLIC_IP}" ]; then
-
-  SECONDS=0
-  END_TIME=$((SECONDS+60))
+# # # # # # # # # # # # # # # # #
+# VARs coming from environment: #
+#                               #
+# EXTERNAL_PUBLIC_IP            #
+# INGRESSGATEWAY_SERVICE_NAME   #
+# PUBLIC_DOMAIN                 #
+# TLS_CERT                      #
+# TLS_KEY                       #
+# # # # # # # # # # # # # # # # #
 
-  while [ ${SECONDS} -lt ${END_TIME} ];do
-      echo "Trying to get loadbalancer IP address"
+function generateXipDomain() {
 
-      EXTERNAL_PUBLIC_IP=$(kubectl get service -n istio-system ${INGRESSGATEWAY_SERVICE_NAME} -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
+    if [ -z "${EXTERNAL_PUBLIC_IP}" ]; then
 
-      if [ "${EXTERNAL_PUBLIC_IP}" ]; then
-          echo "External public IP address is ${EXTERNAL_PUBLIC_IP}"
-          break
-      fi
-      
-      sleep 10
-  done
-fi
+        if [ -z "${INGRESSGATEWAY_SERVICE_NAME}" ]; then
+            INGRESSGATEWAY_SERVICE_NAME=istio-ingressgateway
+        fi
 
-if [ -z "${EXTERNAL_PUBLIC_IP}" ]; then
-    echo "External public IP not found"
-    exit 1
-fi
+        SECONDS=0
+        END_TIME=$((SECONDS+60))
 
-XIP_PATCH_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+        while [ ${SECONDS} -lt ${END_TIME} ];do
+            echo "Trying to get loadbalancer IP address"
 
-PUBLIC_DOMAIN="${EXTERNAL_PUBLIC_IP}.xip.io"
+            EXTERNAL_PUBLIC_IP=$(kubectl get service -n istio-system ${INGRESSGATEWAY_SERVICE_NAME} -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
 
-CERT_PATH="${XIP_PATCH_DIR}/cert.pem"
-KEY_PATH="${XIP_PATCH_DIR}/key.pem"
+            if [ -n "${EXTERNAL_PUBLIC_IP}" ]; then
+                echo "External public IP address is ${EXTERNAL_PUBLIC_IP}"
+                break
+            fi
 
-openssl req -x509 -nodes -days 5 -newkey rsa:4069 \
-                 -subj "/CN=${PUBLIC_DOMAIN}" \
-                 -reqexts SAN -extensions SAN \
-                 -config <(cat /etc/ssl/openssl.cnf \
-        <(printf "\\n[SAN]\\nsubjectAltName=DNS:*.%s" "${PUBLIC_DOMAIN}")) \
-                 -keyout "${KEY_PATH}" \
-                 -out "${CERT_PATH}"
+            sleep 10
+        done
+    fi
 
-TLS_CERT=$(base64 "${CERT_PATH}" | tr -d '\n')
-TLS_KEY=$(base64 "${KEY_PATH}" | tr -d '\n')
+    if [ -z "${EXTERNAL_PUBLIC_IP}" ]; then
+        echo "External public IP not found"
+        exit 1
+    fi
 
-rm "${CERT_PATH}"
-rm "${KEY_PATH}"
+    PUBLIC_DOMAIN="${EXTERNAL_PUBLIC_IP}.xip.io"
 
-PUBLIC_DOMAIN_YAML=$(cat << EOF
+    DOMAIN_YAML=$(cat << EOF
 ---
 data:
   global.domainName: "${PUBLIC_DOMAIN}"
 EOF
 )
 
-TLS_CERT_YAML=$(cat << EOF
+    kubectl patch configmap installation-config-overrides --patch "${DOMAIN_YAML}" -n kyma-installer
+
+}
+
+function generateCerts() {
```

Comment:
Make this function generic, too. Extract generic functions to a separate script that can be source(d).
There will be another script containing logic for application-connector xip setup.
If we manage to somehow reuse THIS script for application-connector (unlikely) we can always merge everything into a single script.